mempool/src/verify.rs

```diff
@@ -274,6 +276,19 @@ pub(crate) async fn verify_tx<'a>(
         recipient_current_balance -= recipient_state.total;
     }
 
+    // The sender must be able to at least pay the fee (in case the tx fails), assumming all pending txns in the mempool for this sender are included in a block
+    if !sender_account.can_fee_be_paid(
+        transaction,
+        sender_current_balance + transaction.fee,
+        blockchain.timestamp(),
+    ) {
+        // If the fee cannot be paid for this transaction, we reject it.
+        return Err(VerifyErr::CantPayFee);
+    }
```

Comment:
This check is designed for vesting/htlc contracts, because apart from just verifying if the fee can be paid from a balance perspective, in the case of vesting/htlc we need to make sure the fee can actually be deducted from the contract (contract already vested with enough funds, etc)